### PR TITLE
Update WorkspaceFileResolver.cpp

### PR DESCRIPTION
**Pull request by Arc157**
This patch fixes issue: https://github.com/JohnnyMorganz/luau-lsp/issues/802

### CODE REVIEW

File: src/WorkspaceFileResolver.cpp [hunk 1]
```diff
@@ -49,8 +49,13 @@ const TextDocument* WorkspaceFileResolver::getTextDocumentFromModuleName(const L
...
 
-    if (auto filePath = platform->resolveToRealPath(name))
-        return getTextDocument(Uri::file(*filePath));
+    if (auto filePath = platform->resolveToRealPath(name)) {
+		auto it = managedFiles.find(filePath->generic_string());
+        if (it != managedFiles.end()) {
+            return &it->second;
+		}
+		return getTextDocument(Uri::file(*filePath));
+	}
```
Comment (checkraisefold) [inline src/WorkspaceFileResolver.cpp:58]: indentation is messed up! use spaces not tabs

Comment (Arc157) [reply src/WorkspaceFileResolver.cpp:58]: Sorry about that, in a bit of a rush

File: src/WorkspaceFileResolver.cpp [hunk 2]
```diff
@@ -49,8 +49,13 @@ const TextDocument* WorkspaceFileResolver::getTextDocumentFromModuleName(const L
     if (Luau::startsWith(name, "untitled:"))
         return getTextDocument(Uri::parse(name));
 
-    if (auto filePath = platform->resolveToRealPath(name))
-        return getTextDocument(Uri::file(*filePath));
+    if (auto filePath = platform->resolveToRealPath(name)) {
```
Comment (checkraisefold) [inline src/WorkspaceFileResolver.cpp:52]: this line uses spaces but the rest are still tabs

**Comment by JohnnyMorganz** [comment]
Thanks for the PR! And so sorry for the delay here.

I've decided to fix this in a separate PR #882 with appropriate test cases

**Pull request closed without merging**